Automattic/simplenote-ios
# Add SimplenoteIntents extension to app

### Fix
This PR lays the ground work for adding intent configuration widgets into Simplenote.

As compared to the new note widget, which displays no data, to be able to setup widgets that can be configured by the user we need to create another extension for the app that defines an Intent configuration.  This PR sets that up.

### Test
There isn't much to test with this PR, but you can confirm that the option to configure a widget is there...
There is a placeholder for the note preview widget that was added to be able to confirm the intents were working, so to test this pr, make sure that you have configuration options.  To do that.
1. run the app on device or in the sim to install the widgets
2. On the home screen long press till the home screen editor appears, tap on the plus
3. find simplenote and select the note preview widget
4. Once it the widget is added to your home screen, while home screen edit is enabled, tap on note preview widget.
5. you will see an option to select the note.  Tap on that and you will be given four choices Note 1-4
6. select one and see that the text appears in the widget.

### Review
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.
